### ISSUE TITLE
strictNullChecks triggers "Variable used before being assigned" when tuple destructuring is used in function parameters

### ISSUE THREAD

**Issue by ntrrgc** [Microsoft/TypeScript#8686]
**TypeScript Version:** 

nightly (1.9.0-dev.20160519)

**Code**

```ts
// A self-contained demonstration of the problem follows...
function test(callback: (tuple: [number, number]) => number) {
    return callback([1, 2]);
}

console.log(test(([a, b]) => a + b));
```

**Expected behavior:**

It should always compile without errors.

**Actual behavior:** 

When `strictNullChecks` is on the following error is raised.
```
Error:(5, 30) TS2454: Variable 'a' is used before being assigned.
Error:(5, 34) TS2454: Variable 'b' is used before being assigned.
```
Obviously it's wrong, as both `a` and `b` are passed as parameters. The signature of `callback` guarantees they have been assigned by the caller.

**Comment by ahejlsberg** [comment]
I just tried the repro and I can't reproduce the problem. I compiles with no errors with a fresh install of nightly build 1.9.0-dev.20160519-1.0.

**Comment by ntrrgc** [comment]
After trying it in several ways I found it was probably fixed in a previous release, but both of my IDEs (IntelliJ and Visual Studio Code) had not picked the update I've installed just before.

After restarting them, it works without error. `tsc` also shows no errors, so I'm closing this. Thanks for the help.
